### ISSUE TITLE
Timeline item resizing is not working when some time has elapsed after pressing the left mouse button

### ISSUE THREAD

**Issue by NorbertSandor** [almende/vis#2098]
Browser: Firefox 48.0

Problem: resizing an item works only if you start resizing immediately (fast enough) after pressing the left mouse button. If you wait some time, resizing won't happen.

Steps to reproduce:
- go to http://visjs.org/examples/timeline/editing/editingItemsCallbacks.html
- select "item 4"
- move to cursor to the left side of "item 4" (the cursor's shape changes to "resize" mode)
- press the left mouse button but do not start resizing
- hold the mouse button, wait some seconds
- now, if you try to resize the item, the cursor simply moves out of the item but resizing won't happen

**Comment by tomzx** [comment]
I could reproduce this with Firefox 43.0.1, 45.0.2, 47.0.1 as well as on Chrome 53.0.2785.116 m, so it does not seem to be browser related.

**Comment by mojoaxel** [comment]
@NorbertSandor Thanks for reporting this and also to @tomzx for confirming this.
I consider this a minor issues, because it does not break anything. Is that a problem for your guys?

**Comment by NorbertSandor** [comment]
I was surprised but it is really confusing/annoying for the end users. The biggest problem is that the timeout is relatively short so it is very easy to run into this error...

**Comment by mojoaxel** [comment]
Ok! Than I recommend trying to increase this timeout ;-)

**Comment by yotamberk** [comment]
@mojoaxel Do you know which timeout? I don't the configuration for the event release of the drag handler...

**Comment by mojoaxel** [comment]
> Do you know which timeout?

There is a [deduplication timout in hammer.js set to 2.5s](https://github.com/hammerjs/hammer.js/commit/bfeb89a77f778c527f771150d1e9687bd318ce8d#diff-d61bfbd2add2f0bf6821207e80705f11R81). 

My thoughts are: Maybe we can add an option to disable Hammer.js if not needed (non touch devices) and fall back on normal mouse events?

**Comment by mojoaxel** [comment]
I think that's a minor issue and would need big changes. Maybe we can even close it?

**Comment by yotamberk** [comment]
@mojoaxel I don't think we should close it. Adding an option of removing Hammer.js for non-touch devices - using pure mouse events, is a great feature request, but doesn't solve this issue. Increasing the timeout is the way to go in the mean time.
In any case, this issue is not critical. It can wait

**Comment by mojoaxel** [comment]
>  Adding an option of removing Hammer.js for non-touch devices - using pure mouse events, is a great feature request

Removing hammer.js completely is not the way to go, because vis.js seems to depend on it for all input handling. Maybe we should move this issue over to hammer.js. I'm sure we can create a nice little example without vis.js that demonstrates the problem on non-touch devices.

> Increasing the timeout is the way to go in the meantime.

We can't increase the timeout! This has to be done in hammer.js. Also an Increase would not get rid of the problem.

**Comment by yotamberk** [comment]
>Maybe we should move this issue over to hammer.js. I'm sure we can create a nice little example without vis.js that demonstrates the problem on non-touch devices.

Agreed. I see, though, that hammerjs has not been maintained for the last few months...

>We can't increase the timeout! This has to be done in hammer.js. Also an Increase would not get rid of the problem.

I hoped there was a way to define the timeout externally... I see now that it is indeed not possible... And in any case it really doesn't solve the problem... 

I'm going to try to implement the `panmove` with `mousedown` and `mousemove` events. Maybe we can solve this with pure mouse events (at least for non-touch devices)

**Comment by adriende** [comment]
I have the same problem, I want to increase the time between the moment when we click ant the moment when we move the mouse.
Do you have a solution ?

**Comment by bradh** [comment]
@adriende: The history is there. I don't see a solution in that though.

**Comment by FrankW76** [comment]
> We can't increase the timeout!

I do not agree with this statement.
This timeout is configurable in Hammer.js. See http://hammerjs.github.io/recognizer-press/

I succeeded in changing this timeout by adding some lines in the visjs library and recompiling.
I would however like this to be added to the options in visjs. I tried to do this, but my JS knowledge is to limited to achieve this (GWT programmer).

Anyhow. These are the changes I did in VisJS to change the timeout to 10000ms. I just recompiled, and put the newly generated vis.js file in my project and it worked as I wanted.

I added the following line in 2 places :
> this.hammer.get('press').set({time:10000});

1. timeline/Core.js 
line 140 
(below the line _this.hammer.get('pan').set({threshold:5, direction: Hammer.DIRECTION_HORIZONTAL});_)
2. timeline/component/ItemSet.js 
line 266 
(below the line _this.hammer.get('pan').set({threshold:5, direction: Hammer.DIRECTION_HORIZONTAL});_)


I also don't agree that this is minor issue. Almost all of my users (which are +60 year old and not tech savvy to be clear) are having a lot of trouble when trying to resize items in the timeline.

**Comment by yotamberk** [comment]
@FrankW76 I'll will try to add this. Thanks for pointing out the option. I don;t know how I missed it in the first place (maybe it didn't exist at the time of the comment?)
Anyways, I'm on it

**Comment by jonas-lauber** [comment]
Hi all,
Gott.. I spent 2 hours trying to figure out why resize wasn't working until I found this issue and figured out I wan't quick enough ^^
Anyway, thanks for the fix @FrankW76. Here is how you can patch the problem in vis.js without recompiling it ;-)
Before to create a new instance of the Timeline, do:

> import * as Hammer from 'vis/lib/module/hammer'; // Need to get the Hammer's version from vis.js
> Hammer.defaults.preset[6].push({time: 10000});

Or you can do it in one line:

> require('vis/lib/module/hammer').defaults.preset[6].push({time: 10000});

This will set a default time for the PressRecognizer in Hammer.js. 

Hope this helps.
